### DISCUSSION TITLE
Question - OPA Policy for restricting TCP Traffic

### DISCUSSION THREAD

**Discussion by robincher** [open-policy-agent/community#220] [OPA and Rego] [answered]
Hi,

I am pretty new to OPA,so just wondering if its possible for OPA to restrict TCP traffic and its request limit? If so, i wonder if there are any examples or recipe?

For example, we want to restrict the file limit a ssh client can pull from the server.

Thanks!

**Comment by srenatus** [comment, marked as answer]
Hello, welcome to OPA!

I'm afraid there are a few things to unpack here: OPA in itself is not an authorization solution for TCP traffic. I.e. it's not a proxy of any sort, but a general purpose policy engine.

As such, it can plugged into many things, and I'm sure there's something that'll make this possible. What's your environment looking like? Have you found anything on the [Ecosystem page](https://www.openpolicyagent.org/docs/latest/ecosystem/) that seems useful to you?

**Reply by robincher** [reply]
Hi, just read through the ecosystem page, and i guess the closest i can checkout is Envoy based on my use case that run on Kong Mesh/Kuma.
